docs/en/development/developer-instruction.md

```diff
@@ -102,29 +102,17 @@ Below you can find some quick links which may be useful when writing code for Cl
 
 ### IDE {#ide}
 
-**CLion (recommended)**
+[Visual Studio Code](https://code.visualstudio.com/) and [Neovim](https://neovim.io/) are two options that have worked well in the past for developing ClickHouse. If you are using VS Code, we recommend using the [clangd extension](https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.vscode-clangd) to replace IntelliSense as it is much more performant.
 
-If you do not know which IDE to use, we recommend that you use [CLion](https://www.jetbrains.com/clion/).
-CLion is commercial software, but it offers a 30 day free trial.
-It is also free of charge for students.
-CLion can be used on both Linux and macOS.
+[KDevelop](https://kdevelop.org/) and [Qt Creator](https://www.qt.io/product/development-tools) are other great alternatives. While KDevelop is a great IDE, it is sometimes unstable. If KDevelop crashes when opening a project, you should click the "Stop All" button as soon as it has opened the list of project files. After doing so, KDevelop should be fine to work with.
```

Comment:
KDevelop does not work for many years, it is crashing almost immediately, and there is no workaround.